### PR TITLE
[Keymap] add coldwater planck keymap

### PR DESCRIPTION
**Pull request by coldwateryu**
## Description
Adding a new keymap for planck keyboard. I believe the most items in checklist below apply so I left them unchecked. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+#define TDPLNX TD(TD_PL_NX)
+#define TDSTPR TD(TD_ST_PR)
+#define TDLCCA TD(TD_LC_CA)
+#define TDF4AF4 TD(TD_F4_ALTF4)
+#define TDCSCL TD(TD_C_SCL)
+#define TDDCL TD(TD_D_CL)
+#define TOGGLE TD(TD_TOGGL)
+#define VD_RGHT LGUI(LCTL(KC_RGHT))
+#define VD_LEFT LGUI(LCTL(KC_LEFT))
+#define xxxxx KC_NO
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]: This is already defined in QMK core, as `XXXXXXX`.

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+	KC_LCTL,	KC_LWIN,  KC_LALT,  KC_LSFT,  xxxxx,    xxxxx,  	xxxxx,    xxxxx,    TDF4AF4,  KC_F5,    KC_F6,    KC_F11,	\
+	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    KC_F1,    KC_F2,    KC_F3,    KC_F12,	\
+	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx	\
+),
+
+[_MOUSE] = LAYOUT_planck_grid(
+	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	KC_WH_U,  KC_WH_L,  KC_MS_U,  KC_WH_R,  xxxxx,   \
+	KC_LCTL,	KC_LWIN,  KC_LALT,  KC_LSFT,  xxxxx,    xxxxx,    xxxxx,  	KC_WH_D,  KC_MS_L,  KC_MS_D,  KC_MS_R,  KC_LCTL, \
+	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,   \
+	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    KC_BTN1,  KC_BTN2,  xxxxx,    xxxxx,    xxxxx \
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
  KC_Q,     KC_W,     KC_F,     KC_P,     KC_B,     TDSTPR,   TDPLNX,   KC_J,     KC_L,      KC_U,     KC_Y,     TDQMIN,
  CTLA,     GUIR,     ALTS,     SFTT,     KC_G,     KC_BRID,  KC_BRIU,  KC_M,     SFTN,      ALTE,     GUII,     CTLO,
  KC_Z,     KC_X,     KC_C,     KC_D,     KC_V,     KC_VOLD,  KC_VOLU,  KC_K,     KC_H,      TDCSCL,   TDDCL,    TDSLBS,
  VD_LEFT,    xxxxx,    MSESC,    LOWBSP,   KC_LSFT,  KC_DEL,   KC_TAB,   KC_SPC,   RAISENT,   FNTAB,    xxxxx,    VD_RGHT
),

[_LOWER] = LAYOUT_planck_grid(
  KC_GRV,   xxxxx,    KC_LPRN,  KC_RPRN,  xxxxx,    xxxxx,    xxxxx,    xxxxx,    KC_7,     KC_8,     KC_9,     KC_EQL,
  TDLCCA, 	KC_LWIN,  ALLP,     SFRP,     KC_TAB,   xxxxx,    xxxxx,    xxxxx,    SF4,      AL5,      GUI6,     CMIN,
  CZ,       CX,       CC,       KC_PSCR,  CV,       xxxxx,    xxxxx,    xxxxx,    KC_1,     KC_2,     KC_3,     KC_SLSH,
  xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    KC_0,     xxxxx,    KC_DOT,   xxxxx
),

[_RAISE] = LAYOUT_planck_grid(
	DIVVY,    F_CHR,    F_WIN,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    KC_PGUP,	KC_HOME,  KC_UP,    KC_END,   xxxxx,
	KC_LCTL,	KC_LWIN,  KC_LALT,  KC_LSFT,  xxxxx,    xxxxx,    xxxxx,    KC_PGDN,  KC_LEFT,  KC_DOWN,  KC_RGHT,	KC_INS,
	xxxxx,  	KC_X,     xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,   	xxxxx,
	xxxxx,  	xxxxx,    _______,  _______,  xxxxx,    KC_TAB,   xxxxx,  	KC_SPC,    xxxxx,    xxxxx,    xxxxx,  	xxxxx
),

[_FUNCT] = LAYOUT_planck_grid(
	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    KC_F7,    KC_F8,    KC_F9,    KC_F10,
	KC_LCTL,	KC_LWIN,  KC_LALT,  KC_LSFT,  xxxxx,    xxxxx,  	xxxxx,    xxxxx,    TDF4AF4,  KC_F5,    KC_F6,    KC_F11,
	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    KC_F1,    KC_F2,    KC_F3,    KC_F12,
	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx
),

[_MOUSE] = LAYOUT_planck_grid(
	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	KC_WH_U,  KC_WH_L,  KC_MS_U,  KC_WH_R,  xxxxx,
	KC_LCTL,	KC_LWIN,  KC_LALT,  KC_LSFT,  xxxxx,    xxxxx,    xxxxx,  	KC_WH_D,  KC_MS_L,  KC_MS_D,  KC_MS_R,  KC_LCTL,
	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,
	xxxxx,  	xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    xxxxx,    KC_BTN1,  KC_BTN2,  xxxxx,    xxxxx,    xxxxx
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+)
+
+};
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+      return false;
+      break;
+    case BACKLIT:
+      if (record->event.pressed) {
+        register_code(KC_RSFT);
+        #ifdef BACKLIGHT_ENABLE
+          backlight_step();
+        #endif
+        #ifdef KEYBOARD_planck_rev5
+          PORTE &= ~(1<<6);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
          writePinLow(E6);
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+        #endif
+      } else {
+        unregister_code(KC_RSFT);
+        #ifdef KEYBOARD_planck_rev5
+          PORTE |= (1<<6);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
          writePinHigh(E6);
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+        muse_tempo+=1;
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      #ifdef MOUSEKEY_ENABLE
+        register_code(KC_MS_WH_DOWN);
+        unregister_code(KC_MS_WH_DOWN);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
        tap_code(KC_WH_D);
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+      #else
+        register_code(KC_PGDN);
+        unregister_code(KC_PGDN);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
        tap_code(KC_PGDN);
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+      #endif
+    } else {
+      #ifdef MOUSEKEY_ENABLE
+        register_code(KC_MS_WH_UP);
+        unregister_code(KC_MS_WH_UP);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
        tap_code(KC_WH_U);
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 9]
```diff
@@ -0,0 +1,411 @@
...
+      #else
+        register_code(KC_PGUP);
+        unregister_code(KC_PGUP);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
        tap_code(KC_PGUP);
```

File: keyboards/planck/keymaps/coldwater/keymap.c [hunk 10]
```diff
@@ -0,0 +1,411 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "muse.h"
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline keyboards/planck/keymaps/coldwater/keymap.c]:
```suggestion
```

`keymap_config` should already be visible to this file.

**Comment by coldwateryu** [comment]
thanks @fauxpark, I'll clean it up before creating a new pull request.

**Comment by fauxpark** [comment]
You don't need to close this PR, just commit and push to the same branch and your changes will show up here.

**Comment by coldwateryu** [comment]
I'm aware, but your comments indicate just how much I can improve my setup, which I'll do before creating the new pull request.

**Pull request closed without merging**